src/app/terminal.js

```diff
@@ -119,13 +124,31 @@ class Terminal {
                 this.broadcastTimeAccumulator >= this.nextBroadcastTime;
         };
 
-        // If it's time to initiate a RTS, do so
-        this.rtsTimeAccumulator += deltaTime;
+        if(this.sentData) {
+            if(!this.currentBroadcast && !this.channelIsBusy()) {
+                this.dataTimeAccumulator+=deltaTime;                
+                if(this.dataAttempts>=numberOfAttempts) {
+                    this.sentData = false;
+                    this.quietModeTime=0;
+                    this.someoneElseHasCts = false;
+                    this.dataTimeout=dataSizeCoefficient*2;
+                }
+                else if(this.dataTimeAccumulator>=this.dataTimeout) {
+                    this.broadcastQueue=[];
```

Comment:
You're modifying what `this.broadcastQueue` points to instead of emptying the array in-place. I'm not sure how well this will work with it being passed as default parameter.